src/server/req_runjob.c

```diff
@@ -278,6 +278,60 @@ call_to_process_hooks(struct batch_request *preq, char *hook_msg, size_t msg_len
 	return rc;
 }
 
+/**
+ * @brief
+ * 	need_to_run_elsewhere - Check whether this request needs to run in another server.
+ *
+ * @param[in] preq - pointer to batch request structure
+ *
+ * @return bool
+ * @retval true - request needs to run in another server
+ * @retval false - request can be executed by self
+ *
+ */
+bool
+need_to_run_elsewhere(struct batch_request *preq)
+{
+	char destination[MAX_SVR_ID + 1];
+	char *pc;
+
+	if (!preq->rq_extend)
+		return false;
+
+	strncpy(destination, preq->rq_extend, sizeof(destination) - 1);
+	if ((pc = strchr(destination, ':')) != NULL)
+		*pc = '\0';
+	if (strcmp(destination, pbs_conf.pbs_server_name))
```

Comment:
What happens if pbs_conf has other variables to tell the server name which overrides PBS_SERVER_NAME?